jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/CountingCallback.java

```diff
@@ -67,23 +67,16 @@ public void succeeded()
     }
 
     @Override
-    public void failed(Throwable failure)
+    public boolean abort(Throwable cause)
     {
-        // Forward failure on the first failure.
-        while (true)
-        {
-            int current = count.get();
-
-            // Already completed ?
-            if (current == 0)
-                return;
+        return count.updateAndGet(c -> c > 1 ? 1 : 0) == 1 && super.abort(cause);
```

Comment:
If `c==1`, it should stay at 1, so:

```suggestion
        return count.updateAndGet(c -> c >= 1 ? 1 : 0) == 1 && super.abort(cause);
```